chain/chunks/src/lib.rs

```diff
@@ -730,34 +732,28 @@ impl ShardsManager {
                     let epoch_id = unwrap_or_return!(
                         runtime_adapter.get_epoch_id_from_prev_block(known_header.prev_hash())
                     );
-                    let block_producer =
-                        unwrap_or_return!(runtime_adapter.get_block_producer(&epoch_id, height));
-                    if runtime_adapter
-                        .verify_validator_signature(
-                            &epoch_id,
-                            &known_header.prev_hash(),
-                            &block_producer,
-                            header.chunk_hash().as_ref(),
-                            header.signature(),
-                        )
-                        .unwrap_or(false)
-                    {
+                    if runtime_adapter.verify_chunk_header_signature(&header, &epoch_id, known_header.prev_hash()).unwrap_or(false) {
                         // We prove that this one is valid for `epoch_id`.
                         // We won't store it by design if epoch is changed.
-                        *stored_chunk = partial_encoded_chunk.clone();
+                        if stored_chunk.len() < MAX_STORED_PARTIAL_CHUNK_SIZE {
+                            stored_chunk.push(partial_encoded_chunk.clone());
```

Comment:
Is this attackable?

Reply:
Yeah I think this is not optimal because the validation here only validate the header for the PartialEncodedChunk, so it's possible that someone sends a lot of invalid PartialEncodedChunk messages and saturate this temporary storage. However, these PartialEncodedChunks will be found invalid when the block is ready and the node calls `process_partial_encoded_chunks`. It will not cause any real problem, but delays of block processing. 

I think I can move more validation logic in `process_partial_encoded_chunk` to here, or even just calls `process_parital_encoded_chunk` here instead of simply storing it, but I want to do it in another PR.